File: CollatzConjecture.ex
defmodule Collatz do
  def hotpo(n) do
    conject(0, n)
  end
  
  def conject(acc, 1) do
    acc
  end
  
  def conject(acc, num) do
    case rem(num, 2) do
      0 -> conject(acc+1, div(num, 2))
      1 -> conject(acc+1, 3 * num + 1)
    end
  end
end

File: CollatzConjecture.hs
collatz :: Int -> String
collatz 1 = show 1
collatz x = conject x ""

conject :: Int -> String -> String
conject x y
  | x == 1 = y ++ (show x)
  | otherwise = if even x then conject (x `div` 2) (y ++ show x ++ "->") 
                          else conject (x*3 + 1) (y ++ show x ++ "->")
